src/resources/cache/hashes.json

```diff
@@ -0,0 +1,106 @@
+{
+  "webgpu/shader/execution/binary/af_addition.bin": "593c077",
+  "webgpu/shader/execution/binary/af_logical.bin": "590d559",
+  "webgpu/shader/execution/binary/af_division.bin": "5931d73",
+  "webgpu/shader/execution/binary/af_matrix_addition.bin": "593a5c8",
+  "webgpu/shader/execution/binary/af_matrix_subtraction.bin": "5935065",
+  "webgpu/shader/execution/binary/af_multiplication.bin": "59384d1",
+  "webgpu/shader/execution/binary/af_remainder.bin": "5938a00",
+  "webgpu/shader/execution/binary/af_subtraction.bin": "593f79c",
+  "webgpu/shader/execution/binary/f16_addition.bin": "593d697",
+  "webgpu/shader/execution/binary/f16_logical.bin": "590a70a",
+  "webgpu/shader/execution/binary/f16_division.bin": "593d64d",
+  "webgpu/shader/execution/binary/f16_matrix_addition.bin": "5930c81",
+  "webgpu/shader/execution/binary/f16_matrix_matrix_multiplication.bin": "5932f2b",
+  "webgpu/shader/execution/binary/f16_matrix_scalar_multiplication.bin": "593b6f3",
+  "webgpu/shader/execution/binary/f16_matrix_subtraction.bin": "593bd88",
+  "webgpu/shader/execution/binary/f16_matrix_vector_multiplication.bin": "5936d87",
+  "webgpu/shader/execution/binary/f16_multiplication.bin": "5931679",
+  "webgpu/shader/execution/binary/f16_remainder.bin": "593ae6d",
+  "webgpu/shader/execution/binary/f16_subtraction.bin": "5932799",
+  "webgpu/shader/execution/binary/f32_addition.bin": "593df7c",
+  "webgpu/shader/execution/binary/f32_logical.bin": "5901281",
+  "webgpu/shader/execution/binary/f32_division.bin": "593b194",
+  "webgpu/shader/execution/binary/f32_matrix_addition.bin": "5937c46",
+  "webgpu/shader/execution/binary/f32_matrix_matrix_multiplication.bin": "5931426",
+  "webgpu/shader/execution/binary/f32_matrix_scalar_multiplication.bin": "593ff67",
+  "webgpu/shader/execution/binary/f32_matrix_subtraction.bin": "59328ea",
+  "webgpu/shader/execution/binary/f32_matrix_vector_multiplication.bin": "5939a29",
+  "webgpu/shader/execution/binary/f32_multiplication.bin": "593aa04",
+  "webgpu/shader/execution/binary/f32_remainder.bin": "593e328",
+  "webgpu/shader/execution/binary/f32_subtraction.bin": "593c524",
+  "webgpu/shader/execution/binary/i32_arithmetic.bin": "5914697",
```

Comment:
These hashes are very poorly distributed. Is that a problem?

Reply:
It was my shoddy `hashCombine` which was shifting right then xor-ing. This meant the MSBs of the hash were always taken from the last import. Fixed by CRC'ing the string of the two hashes.